table/table.cpp

```diff
@@ -164,59 +133,120 @@ void Table::edit_rows(vector<string> &columns,
             Info index = column_index(column);
             edit_indecies.push_back(index);
         } catch (std::exception e) {
-            // catch code
-            // throw()
+            throw e;
         }
     }
+
+    // Loops through rows and edits the ones that fit comparisons
     for(size_t i = 0; i < table.rows_size(); ++i) {
-        bool good = true;
-        for(size_t j = 0; j < comparisons.size(); ++j) {
-            bool correct_entry = correct_type(indecies[j], table.rows(i).entries(indecies[j].index));
-            bool correct_comparison = correct_type(indecies[j], comparisons[j]);
-            if(!correct_entry || !correct_comparison) {
-                // throw()
-            }
-            if(!compare_entries(table.rows(i).entries(indecies[j].index), comparisons[j])) {
-                good = false;
-                j = comparisons.size();
-            }
-        }
+        bool good = check_row(table.rows(i), indecies, comparisons);
+
         if(!good) {
             continue;
         }
+
+        // Edits each specified entry in row
         for(size_t j = 0; j < entries.size(); ++j) {
             if(!correct_type(edit_indecies[j], entries[j])) {
-                // throw
+                throw type_mismatch(type_mismatch::error_type::updation, get_type(edit_indecies[j]), get_type(entries[j]));
             }
             ::database::Entry* edit_entry = table.mutable_rows(i)->mutable_entries(edit_indecies[j].index);
             edit_entry->CopyFrom(entries[j]);
         }
     }
 } // edit_rows()
 
-void Table::insert(::database::Row &row) {
-    for(auto it : column_indecies) {
+// Edits all rows to set entry@each edit_column = the said new entry
+// Input: vector<string> &edit_columns -> columns being edited in filtered entries
+// Input: vector<::database::Entry> &entries -> new entries to update entries in edit_columns
+void Table::edit_all(vector<string> &edit_columns, 
+                vector<::database::Entry> &entries) {
+    vector<Info> edit_indecies;
+    for (string column: edit_columns) {
```

Comment:
nit: Please add a space between the looping variable and the color in a range-based for loop

i.e.

`for (string column : edit_columns) {`